dds/DCPS/XTypes/DynamicDataHolder.h

```diff
@@ -0,0 +1,807 @@
+  {
+    return DDS::RETCODE_UNSUPPORTED;
+  }
+
+  DDS::ReturnCode_t get_char8_value(CORBA::Char& value,
+                                    DDS::MemberId id)
+  {
+    DDS::DynamicTypeMember_var member;
+    if (type_->get_member(member, id) != DDS::RETCODE_OK) {
+      return DDS::RETCODE_ERROR;
+    }
+
+    DDS::MemberDescriptor_var md;
+    if (member->get_descriptor(md) != DDS::RETCODE_OK) {
+      return DDS::RETCODE_ERROR;
+    }
+
+    DDS::DynamicType_var type = get_base_type(md->type());
+    if (type->get_kind() != TK_CHAR8) {
+      return DDS::RETCODE_ERROR;
+    }
+
+    const DCPS::Value v = meta_struct_.getValue(&value_, id);
+    value = v.c_;
+    return DDS::RETCODE_OK;
+  }
+
+  DDS::ReturnCode_t set_char8_value(DDS::MemberId id,
+                                    CORBA::Char value)
+  {
+    return DDS::RETCODE_UNSUPPORTED;
+  }
+
+  DDS::ReturnCode_t get_char16_value(CORBA::WChar& value,
+                                     DDS::MemberId id)
+  {
+    // TODO: Help.
+    return 0;
+  }
+
+  DDS::ReturnCode_t set_char16_value(DDS::MemberId id,
+                                     CORBA::WChar value)
+  {
+    return DDS::RETCODE_UNSUPPORTED;
+  }
+
+  DDS::ReturnCode_t get_byte_value(CORBA::Octet& value,
+                                   DDS::MemberId id)
+  {
+    // TODO: Help.
+    return 0;
+  }
+
+  DDS::ReturnCode_t set_byte_value(DDS::MemberId id,
+                                   CORBA::Octet value)
+  {
+    return DDS::RETCODE_UNSUPPORTED;
+  }
+
+  DDS::ReturnCode_t get_boolean_value(CORBA::Boolean& value,
+                                      DDS::MemberId id)
+  {
+    DDS::DynamicTypeMember_var member;
+    if (type_->get_member(member, id) != DDS::RETCODE_OK) {
+      return DDS::RETCODE_ERROR;
+    }
+
+    DDS::MemberDescriptor_var md;
+    if (member->get_descriptor(md) != DDS::RETCODE_OK) {
+      return DDS::RETCODE_ERROR;
+    }
+
+    DDS::DynamicType_var type = get_base_type(md->type());
+    if (type->get_kind() != TK_BOOLEAN) {
+      return DDS::RETCODE_ERROR;
+    }
+
+    const DCPS::Value v = meta_struct_.getValue(&value_, id);
+    value = v.b_;
+    return DDS::RETCODE_OK;
+  }
+
+  DDS::ReturnCode_t set_boolean_value(DDS::MemberId id,
+                                      CORBA::Boolean value)
+  {
+    return DDS::RETCODE_UNSUPPORTED;
+  }
+
+  DDS::ReturnCode_t get_string_value(char*& value,
+                                     DDS::MemberId id)
+  {
+    DDS::DynamicTypeMember_var member;
+    if (type_->get_member(member, id) != DDS::RETCODE_OK) {
+      return DDS::RETCODE_ERROR;
+    }
+
+    DDS::MemberDescriptor_var md;
+    if (member->get_descriptor(md) != DDS::RETCODE_OK) {
+      return DDS::RETCODE_ERROR;
+    }
+
+    DDS::DynamicType_var type = get_base_type(md->type());
+    if (type->get_kind() != TK_STRING8) {
+      return DDS::RETCODE_ERROR;
+    }
+
+    const DCPS::Value v = meta_struct_.getValue(&value_, id);
+    // TODO: Allocate?  Transfer?
```

Comment:
I only need string for my purposes.

Reply:
It's an `inout string` parameter. As the implementer, you get a buffer that the caller allocated (it may not be NULL). You can decide to leave it as-is, change the contents of the buffer, or free it and allocate a new one in its place.  In all cases, on return the caller is responsible for the buffer that it points to.  String_var has support for this.